File: python-service/app/services/production_label.py
# Part of the Chili3d Project, under the AGPL-3.0 License.
# See LICENSE file in the project root for full license information.

"""Belt production release label helpers (Track 3a)."""

from __future__ import annotations

from app.services.print_recipe import HARDNESS_TO_INFILL_PCT, PrintRecipeV1


def build_production_release_label(
    *,
    side: str | None,
    recipe: PrintRecipeV1,
    profile_display_name: str,
    client_label: str | None = None,
    design_id: str | None = None,
) -> str:
    side_text = "Left" if side == "left" else "Right" if side == "right" else "Side"
    table = recipe.hardness_to_infill_pct or HARDNESS_TO_INFILL_PCT
    pct = table.get(recipe.default_hardness, HARDNESS_TO_INFILL_PCT[recipe.default_hardness])
    parts = [
        "OrthoCAD belt release",
        side_text,
        recipe.default_hardness,
        f"{pct}% gyroid (experimental)",
        profile_display_name,
    ]
    if client_label and client_label.strip():
        parts.append(client_label.strip())
    if design_id:
        parts.append(f"design {design_id[:8]}")
    return " · ".join(parts)

File: vertex/shared/print-recipe/production-label.ts
// Part of the Chili3d Project, under the AGPL-3.0 License.
// See LICENSE file in the project root for full license information.

import { HARDNESS_TO_INFILL_PCT, type PrintRecipeV1 } from "./print-recipe";

export interface ProductionReleaseLabelInput {
    side: "left" | "right";
    recipe: PrintRecipeV1;
    profileDisplayName?: string;
    clientLabel?: string;
    designId?: string;
}

/**
 * Human-readable belt release label embedded in G-code comments and manufacturing metadata.
 * Not a Shore durometer claim — hardness is the named gyroid ladder only.
 */
export function buildProductionReleaseLabel(input: ProductionReleaseLabelInput): string {
    const side = input.side === "left" ? "Left" : "Right";
    const hardness = input.recipe.defaultHardness;
    const table = input.recipe.hardnessToInfillPct ?? HARDNESS_TO_INFILL_PCT;
    const pct = table[hardness] ?? HARDNESS_TO_INFILL_PCT[hardness];
    const profile = input.profileDisplayName ?? input.recipe.profileId;
    const client = input.clientLabel?.trim();
    const parts = ["OrthoCAD belt release", side, hardness, `${pct}% gyroid (experimental)`, profile];
    if (client) parts.push(client);
    if (input.designId) parts.push(`design ${input.designId.slice(0, 8)}`);
    return parts.join(" · ");
}
